Alicerce/Sources/Persistence/DiskMemoryPersistence.swift

```diff
@@ -0,0 +1,326 @@
+            }
+
+            return completion { () }
+        }
+
+        writeOperationQueue.addOperation(removeOperation)
+    }
+
+    private func calculateUsedDiskSize() {
+
+        let calculateUsedSizeOperation = DiskMemoryBlockOperation(qos: .userInteractive) { [weak self] in
+            guard let `self` = self else { return }
+
+            let urls = self.directoryContents(with: [.fileSizeKey])
+
+            let fileSizes = urls.map {
+                UInt64($0.resourceValues(forKeys: [.fileSizeKey]).fileSize ?? 0)
+            }
+
+            self.usedDiskSize = fileSizes.reduce(UInt64(0), +)
+        }
+
+        readOperationQueue.addOperation(calculateUsedSizeOperation)
+    }
+
+    private func diskPath(for key: Persistence.Key) -> String {
+        return "\(configuration.path)/\(key)"
+    }
+
+    private func hasDiskCacheDirectory() -> Bool {
+        return fileManager.fileExists(atPath: configuration.path)
+    }
+
+    // FIX: Should this error be propagated to the top level?
+    private func createDiskCacheDirectory() -> Bool {
+        do {
+            try fileManager.createDirectory(atPath: configuration.path, withIntermediateDirectories: true, attributes: nil)
+        }
+        catch let error {
+            assertionFailure("failed to create directory `\(configuration.path)` with error: \n\(error)")
+            return false
+        }
+
+        return true
+    }
+
+    private func directoryContents(with keys: [URLResourceKey]) -> [URL] {
+        guard let url = URL(string: configuration.path) else {
+            assertionFailure("💥 failed to create url with path 👉 `\(configuration.path)`")
+            return []
+        }
+
+        guard let urls = try? fileManager.contentsOfDirectory(at: url,
+                                                              includingPropertiesForKeys: keys,
+                                                              options: .skipsPackageDescendants)
+            else {
+                assertionFailure("💥 failed to read directory content for path 👉 \(url.absoluteString)")
+                return []
+        }
+
+        return urls
+    }
+
+    // MARK: - Operations
+
+    private func createEvictOperation() -> DiskMemoryBlockOperation {
+        return DiskMemoryBlockOperation(qos: .background) { [unowned self] in
+
+            // Check if should run eviction
+            guard self.configuration.diskLimit < self.usedDiskSize else {
+                return
+            }
+
+            let extraOccupiedDiskSize = self.usedDiskSize - self.configuration.diskLimit
+
+            let urls = self.directoryContents(with: [.contentAccessDateKey, .fileSizeKey])
+
+            typealias FileAccessTimeSizeTuple = (accessTime: TimeInterval, size: UInt64)
+            typealias FilePathAttributesTuple = (path: String, fileAttr: FileAccessTimeSizeTuple)
+
+            let accessTimes: [FilePathAttributesTuple] = urls.map {
```

Comment:
Is `accessTimes` the best name for this variable? 🤔 It seems to store more than that 😝